core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -113,13 +114,15 @@ private HTTPClient(
       ObjectMapper objectMapper,
       Map<String, String> properties,
       HttpClientConnectionManager connectionManager,
-      AuthSession session) {
+      AuthSession session,
+      OpenTelemetry openTelemetry) {
     this.baseUri = baseUri;
     this.baseHeaders = baseHeaders;
     this.mapper = objectMapper;
     this.authSession = session;
 
-    HttpClientBuilder clientBuilder = HttpClients.custom();
+    HttpClientBuilder clientBuilder =
+        ApacheHttpClientTelemetry.builder(openTelemetry).build().newHttpClientBuilder();
```

Comment:
can we only set openTelemetry when there is something set via builder, i understand default value is NoOp but it would still add interceptor to call of HTTP but i guess it will do nothing, having set by builder would  make sure its only added when required, WDYT ? 

```suggestion
        openTelemetry ? ApacheHttpClientTelemetry.builder(openTelemetry).build().newHttpClientBuilder() : HttpClients.custom();
```